allennlp/data/token_indexers/openai_transformer_byte_pair_indexer.py

```diff
@@ -0,0 +1,179 @@
+from typing import Dict, List, Tuple
+import json
+import tarfile
+
+from overrides import overrides
+
+from allennlp.common.file_utils import cached_path
+from allennlp.common.params import Params
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.token_indexers.token_indexer import TokenIndexer
+
+
+@TokenIndexer.register("openai_transformer_byte_pair")
+class OpenaiTransformerBytePairIndexer(TokenIndexer[List[int]]):
+    """
+    This is unlike most of our TokenIndexers in that its
+    indexing is not based on a `Vocabulary` but on a fixed
+    set of mappings that are loaded by the constructor.
+    """
+    # pylint: disable=no-self-use
+    def __init__(self,
+                 encoder: Dict[str, int],
+                 byte_pairs: List[Tuple[str, str]],
+                 n_ctx: int = 512) -> None:
+
+        self.encoder = encoder
+        self.decoder = {word_id: word for word, word_id in self.encoder.items()}
+
+        # Compute ranks
+        self.bpe_ranks = {pair: idx for idx, pair in enumerate(byte_pairs)}
+
+        self.cache: Dict[str, List[int]] = {}
+        self.n_ctx = n_ctx
+
+    @classmethod
+    def from_params(cls, params: Params) -> 'OpenaiTransformerBytePairIndexer':
```

Comment:
Would it be easier to separate the BPE encoder part from the weights, so we can remove this custom from_params?

Reply:
you know what, I can remove it anyway, the same way I did in the other one.

(I'd prefer just having one tarfile required to run the model, rather than two)